vital/tests/test_sfm_constraints.cxx

```diff
@@ -0,0 +1,297 @@
+
+  // Expect failure if the metadata map is not set
+  EXPECT_FALSE(constraints.get_camera_orientation_prior_local(0, R_loc));
+  EXPECT_EQ(R_loc, rotation_d{});
+
+  // Create metadata map
+  metadata_map::map_metadata_t mdm;
+  add_metadata<VITAL_META_PLATFORM_HEADING_ANGLE>(mdm, 0, 90.0);
+  add_metadata<VITAL_META_PLATFORM_ROLL_ANGLE>(mdm, 0, 90.0);
+  add_metadata<VITAL_META_PLATFORM_PITCH_ANGLE>(mdm, 0, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_AZ_ANGLE>(mdm, 0, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_EL_ANGLE>(mdm, 0, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_ROLL_ANGLE>(mdm, 0, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_ROLL_ANGLE>(mdm, 1, 90.0);
+  add_metadata<VITAL_META_PLATFORM_HEADING_ANGLE>(mdm, 2, 90.0);
+  add_metadata<VITAL_META_PLATFORM_ROLL_ANGLE>(mdm, 2, 90.0);
+  add_metadata<VITAL_META_PLATFORM_PITCH_ANGLE>(mdm, 2, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_AZ_ANGLE>(mdm, 2, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_EL_ANGLE>(mdm, 2, 90.0);
+  add_metadata<VITAL_META_SENSOR_REL_ROLL_ANGLE>(mdm, 2, std::nan(""));
+  metadata_map_sptr metadata_map = std::make_shared<simple_metadata_map>(mdm);
+  constraints.set_metadata(metadata_map);
+  EXPECT_EQ(constraints.get_metadata()->size(), 3);
+
+  // Expect failure if the metadata values are not set
+  EXPECT_FALSE(constraints.get_camera_orientation_prior_local(1, R_loc));
+  EXPECT_EQ(R_loc, rotation_d{});
+
+  // Expect failure if a metadata value is nan
+  EXPECT_FALSE(constraints.get_camera_orientation_prior_local(2, R_loc));
+  EXPECT_EQ(R_loc, rotation_d{});
+
+  // Success case
+  EXPECT_TRUE(constraints.get_camera_orientation_prior_local(0, R_loc));
+  EXPECT_NEAR(R_loc.quaternion().coeffs()[0], 0.0446255, 1e-6);
+  EXPECT_NEAR(R_loc.quaternion().coeffs()[1], -0.26865, 1e-6);
+  EXPECT_NEAR(R_loc.quaternion().coeffs()[2], -0.818742, 1e-6);
+  EXPECT_NEAR(R_loc.quaternion().coeffs()[3], 0.505467, 1e-6);
+}
+
+// ----------------------------------------------------------------------------
+TEST(sfm_constraints, get_camera_position_prior_local)
+{
+  sfm_constraints constraints;
+  vector_3d pos_loc(1, 1, 1);
+
+  // Expect failure if the lgcs is not set
+  EXPECT_FALSE(constraints.get_camera_position_prior_local(0, pos_loc));
+  EXPECT_EQ(pos_loc.x(), 1.0);
+  EXPECT_EQ(pos_loc.y(), 1.0);
+  EXPECT_EQ(pos_loc.z(), 1.0);
+
+  static auto geo_conv = kwiver::arrows::proj::geo_conversion{};
+  set_geo_conv(&geo_conv);
+  local_geo_cs lgcs;
+  lgcs.set_origin(geo_point(vector_3d(1, 1, 1), SRID::lat_lon_WGS84));
+  constraints.set_local_geo_cs(lgcs);
+
+  // Expect failure if the metadata map is not set
+  EXPECT_FALSE(constraints.get_camera_position_prior_local(0, pos_loc));
+  EXPECT_EQ(pos_loc.x(), 1.0);
+  EXPECT_EQ(pos_loc.y(), 1.0);
+  EXPECT_EQ(pos_loc.z(), 1.0);
+
+  // Create metadata map
+  metadata_map::map_metadata_t mdm;
+  add_metadata<VITAL_META_SENSOR_LOCATION>(mdm, 0, geo_point(vector_3d(1, 1, 3),
+    SRID::lat_lon_WGS84));
```

Comment:
Since this line goes just over 80 characters, and it's already being split anyway, I'd suggest making the cutoff point a bit earlier in the line, to avoid going too wide. Maybe something like:
```suggestion
  add_metadata<VITAL_META_SENSOR_LOCATION>(
    mdm, 0, geo_point(vector_3d(1, 1, 3), SRID::lat_lon_WGS84));
```